packages/flutter/lib/src/services/system_chrome.dart

```diff
@@ -381,6 +381,14 @@ abstract final class SystemChrome {
       'SystemChrome.setPreferredOrientations',
       _stringify(orientations),
     );
+    _preferredOrientations = orientations;
+  }
+
+  /// Returns the list of device orientations as they were requested in [setPreferredOrientations].
+  ///
+  /// Returns null, if the [setPreferredOrientations] was never called.
+  static List<DeviceOrientation>? getPreferredOrientations() {
+    return _preferredOrientations;
```

Comment:
Don't we have to query the platform here to find out what the current orientations are? Somebody may have changed these on the native side without the framework knowing.